### DISCUSSION TITLE
'AcceleratorConnector' object has no attribute 'distributed_backend' and other issues

### DISCUSSION THREAD

**Discussion by daniel-a-diaz** [optuna/optuna#4040] [Q&A] [answered]
I have a few things I am trying to troubleshoot and would love some help. Specifically I am working with PyTorch Lightning and Optuna for distributed training. Everything with Optuna works great for me on a single GPU, but when I up it to four I get issues. Below are the versions I have installed with conda.
- python 3.10.6
- pytorch 1.12.1
- pytorch_lightning 1.7.7
- optuna 3.0.2

1. I was trying to run the pytorch_lightning_simple.py example and I received the following error.
    Exception has occurred: AttributeError 'AcceleratorConnector' object has no attribute 'distributed_backend'
    It seems this is related to `PyTorchLightningPruningCallback()`. I have just been using PyTorch Lightning's `EarlyStopping()` instead, but it would be nice to try out some of Optuna's pruners

2. The next issues I have deal with distributed training. When I use `strategy=DDPStrategy()` it will get frozen in epoch 0 of the first trial and then I can't even kill the process with ctrl-c

3. When I use `strategy=DDPSpawnStrategy()` it works through the first two trials, but when the third trial starts I get the following error. OSError: [Errno 24] Too many open files: '/tmp/tmp6tekf0nj'

**Comment by nzw0301** [comment, marked as answer]
As addressed in https://github.com/optuna/optuna/issues/3418 Optuna's callback does not support PyTorchLightning 1.7. Note that we supported DDP with PyTorchLightning 1.5.* and RDB.
